instrumentation/mongo/mongo-3.1/library/src/main/java/io/opentelemetry/instrumentation/mongo/v3_1/MongoAttributesExtractor.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.mongo.v3_1;
+
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.DB_MONGODB_COLLECTION;
+import static java.util.Arrays.asList;
+
+import com.mongodb.event.CommandStartedEvent;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import java.util.HashSet;
+import java.util.Set;
+import org.bson.BsonValue;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+class MongoAttributesExtractor implements AttributesExtractor<CommandStartedEvent, Void> {
+  @Override
+  public void onStart(AttributesBuilder attributes, CommandStartedEvent event) {
+    set(attributes, DB_MONGODB_COLLECTION, collectionName(event));
+  }
+
+  @Override
+  public void onEnd(
+      AttributesBuilder attributes,
+      CommandStartedEvent event,
+      @Nullable Void unused,
+      @Nullable Throwable error) {}
+
+  @Nullable
+  String collectionName(CommandStartedEvent event) {
+    if (event.getCommandName().equals("getMore")) {
+      if (event.getCommand().containsKey("collection")) {
+        BsonValue collectionValue = event.getCommand().get("collection");
```

Comment:
Nit: you could avoid double map lookup by not calling `containsKey`:
```suggestion
      BsonValue collectionValue = event.getCommand().get("collection");
      if (collectionValue != null) {
```